src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -236,35 +252,54 @@ private void LogTrace(LogLevel logLevel, IEnumerable<KeyValuePair<string, object
         }
 
         // Makes sure these are done in the correct order. If there are duplicate keys, the last State property wins.
-        private static void ApplyScopeAndStateProperties(IDictionary<string, string> properties, IEnumerable<KeyValuePair<string, object>> state)
+        private static void ApplyScopeAndStateProperties(IDictionary<string, string> properties, IEnumerable<KeyValuePair<string, object>> state, ITelemetry telemetry)
         {
-            ApplyScopeProperties(properties);
+            ApplyScopeProperties(telemetry);
             ApplyProperties(properties, state, true);
         }
 
-        internal static void ApplyProperty(IDictionary<string, string> properties, string key, object value, bool applyPrefix = false)
+        internal void ApplyKnownProperties(IDictionary<string, string> properties, LogLevel logLevel, EventId eventId)
+        {
+            properties[LogConstants.CategoryNameKey] = _categoryName;
+            properties[LogConstants.LogLevelKey] = LogLevelEnumHelper.ToStringOptimized(logLevel);
+            if (eventId != null)
+            {
+                if (eventId.Id != 0)
+                {
+                    properties[LogConstants.EventIdKey] = Convert.ToString(eventId.Id);
+                }
+                if (!string.IsNullOrEmpty(eventId.Name))
+                {
+                    properties[LogConstants.EventNameKey] = eventId.Name;
+                }
+            }
+        }
+        internal static void ApplyProperty(IDictionary<string, string> properties, string key, object objectValue, bool applyPrefix = false)
         {
             // do not apply null values
-            if (value == null)
+            if (objectValue == null)
             {
                 return;
             }
 
             string stringValue = null;
 
             // Format dates
-            Type propertyType = value.GetType();
-            if (propertyType == typeof(DateTime))
+            if (objectValue is string value)
+            {
+                stringValue = value;
+            }
+            if (objectValue is DateTime date)
```

Comment:
Note sure if you intentionally didn't use `else if` here.
```suggestion
            else if (objectValue is DateTime date)
```

Reply:
I missed it.

Reply:
FWIW a switch statement would be much more concise here with the `else if` change in play (I can't see this reviewing latest on GitHub though)